src/extractor/guidance/sliproad_handler.cpp

```diff
@@ -51,27 +51,36 @@ operator()(const NodeID, const EdgeID source_edge_id, Intersection intersection)
     if (intersection.size() <= 2)
         return intersection;
 
-    const auto findNextIntersectionForRoad =
-        [&](const NodeID at_node, const ConnectedRoad &road, NodeID &output_node) {
-            auto intersection = intersection_generator(at_node, road.turn.eid);
-            auto in_edge = road.turn.eid;
-            // skip over traffic lights
-            while (intersection.size() == 2)
+    const auto findNextIntersectionForRoad = [&](
+        const NodeID at_node, const ConnectedRoad &road, NodeID &output_node) {
+        auto intersection = intersection_generator(at_node, road.turn.eid);
+        auto in_edge = road.turn.eid;
+        // skip over traffic lights
+        // if there are more than 10 traffic lights/other items on the road, we don't consider it
+        // for sliproads anymore. This also prevents us from ending up in an endless loop.
```

Comment:
No. We simply want to prevent endless loops. The same argument as above holds here. If we want to detect endless loops here, we would need an `unordered_set` to detect these. There is simply no situation that justifies a detection as sliproad if there is like 10 or more traffic lights in between. Anything but `1` would be unexpected.